### PR TITLE
feat-added-api-services

### PR DESCRIPTION
**Pull request by raman1999**
Added toast functionality and api calls functions for all functionalty

### CODE REVIEW

File: src/Reducer/userData-reducer.jsx [hunk 1]
```diff
@@ -14,9 +14,13 @@ export function userReducer(state, { type, payload }) {
...
         watchLater: payload,
       };
-
-    case "SHOW_TOAST":
-      return { ...state, toastMsg: payload };
+    case "UPDATE_PLAYLIST_AFTER_VIDEO_ACTION":
+      return {
+        ...state,
+        playlists: state.playlists.map((playlist) =>
+          playlist._id === payload._id ? payload : playlist
```
Comment (shubhamkumar648) [inline src/Reducer/userData-reducer.jsx:21]: i like the way used ternary oerator

**Review by shubhamkumar648 — APPROVED** [review]
good use of context and reducer hooks, code looks good

File: src/Services/historyService.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        var { data } = await axios.delete(
+          `/api/user/history/${type === "CLEAR_HISTORY" ? "all" : payload._id}`,
+          postHeader
+        );
+        dispatch({
+          type: "SET_HISTORY",
+          payload: data.history,
+        });
+        toast.success("History updated successfully");
+        break;
```
Comment (rahulcs754) [inline src/Services/historyService.jsx:33]: I think, you have not need to break here

File: src/Services/likedVideosService.jsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        var { data } = await axios.delete(
+          `/api/user/likes/${payload._id}`,
+          postHeader
+        );
+        dispatch({
+          type: "SET_LIKES",
+          payload: data.likes,
+        });
+        toast.success("Removed from liked videos...");
+        break;
```
Comment (rahulcs754) [inline src/Services/likedVideosService.jsx:32]: same above

**Review by rahulcs754 — APPROVED** [review]
The code looks clean and good.
just small suggestion

File: src/Services/historyService.jsx (hunk 2 above)
Comment (raman1999) [reply src/Services/historyService.jsx:33]: Yes, but it''s good practice to follow with switch case

**Pull request merged**